app/validators/employment_details/answers_validator.rb

```diff
@@ -39,6 +43,55 @@ def not_working_details_complete?
       income.date_job_lost.present?
     end
 
+    # :nocov:
+    def validate_employment
+      return unless employed?
+
+      validate_employment_details
+      validate_employment_income
+    end
+    # :nocov:
+
+    def validate_employment_details
+      return unless requires_full_means_assessment?
```

Comment:
Is this the trigger for collecting full employment details? Is it not just `income_above_threshold?`